var/spack/repos/builtin/packages/docbook-xsl/package.py

```diff
@@ -22,6 +22,13 @@ class DocbookXsl(Package):
     def install(self, spec, prefix):
         install_tree('.', prefix)
 
+    def catalog(self):
+        return os.path.join(self.prefix, 'catalog.xml')
+
     def setup_environment(self, spack_env, run_env):
-        catalog = os.path.join(self.prefix, 'catalog.xml')
+        catalog = self.catalog()
         run_env.set('XML_CATALOG_FILES', catalog, separator=' ')
```

Comment:
Why use set here but prepend_path below? Shouldn't they behave the same? Also, what is `separator` needed for?

Reply:
I used `prepend_path` in the new method I added below because it neatly expressed what I wanted to do and I was able to test that it did what I needed it to do.

I asked myself the same question about the code you've called out.   I puzzled about it a bit but don't understand why it is the way that it is and/or who's using it.  I opted to leave sleeping dogs lie.  The `set` is troubling because `XML_CATALOG_FILES` is supposed to be a list of XML Catalogs.  As I read this bit and its doppelgänger in the `docbook-xml` package, you can't safely use both of them.  But, it seems out of scope.

Reply:
FWIW, [here's where I learned about XML Catalogs](http://xmlsoft.org/catalog.html).  There's also [the XML Catalogs home](http://xmlcatalogs.org/) if you're really getting into it.